known-test-failures.md

```diff
@@ -170,4 +171,18 @@ inside of the cluster.
 
 You can also deploy an [Ingress Controller](
 https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/)
-and use layer 7 load balancing instead if you're looking to load-balance HTTP traffic.
\ No newline at end of file
+and use layer 7 load balancing instead if you're looking to load-balance HTTP traffic.
+
+## SchedulerPreemption is not supported
+
+### What's the issue?
+
+Supervisor cluster does not support Pod priorities and preemption.
+
+### What's the impact?
+
+PriorityClass is ignored in the Supervisor cluster.
+
+### Are there workarounds?
+
+No. PriorityClass, if specified, will be ignored.
```

Comment:
Looks good to me.

Would be great if Ben and/or Pranshu also took a look.

Reply:
I think it would be a bit incorrect to say that both priority and preemption are unsupported and PriorityClass is ignored.

Pod priority also impacts the scheduling order of the pods and this works in the Supervisor cluster as well. Essentially, when Pod priority is enabled, the scheduler orders pending Pods by their priority and a pending Pod with higher priority is placed ahead of other pending Pods with lower priority in the scheduling queue. As a result, the higher priority Pod may be scheduled sooner than Pods with lower priority if its scheduling requirements are met. If such Pod cannot be scheduled, scheduler will continue and tries to schedule other lower priority Pods.

This part of PodPriority impacting the scheduling order leverages K8s scheduler's logic of evaluating pods in the order of their priority and hence it works as-is in the SV clusters and PriorityClass is not ignored.

What does not work (or it'd be more accurate to say that what hasn't been evaluated and the behavior is undeterministic) is the preemption of a lower priority pod when a higher priority pod cannot be scheduled. If no Node is found that satisfies all the specified requirements of a Pod, preemption logic is triggered for the pending Pod and it is this "preemption logic" that has not been evaluated in the SV clusters.

Reply:
@pranshujain28 Thanks for this detailed explanation. I would suppose that the reason this has come up is because we fail a test around pre-emption logic. It would be good if we could meet to agree on a crisp explanation. @vshetty Is this also an issue for versions > 1.19?

Reply:
It is an issue > 1.19 as well, the tests were added in 1.19.

Reply:
@corrieb this is an issue for > 1.19 as well. 
@pranshujain28 can you help come up with a small write-up for this?